client/src/main/java/io/pravega/client/netty/impl/ClientConnectionInboundHandler.java

```diff
@@ -59,6 +63,10 @@ public void channelRegistered(ChannelHandlerContext ctx) throws Exception {
         super.channelRegistered(ctx);
         Channel c = ctx.channel();
         channel.set(c);
+        isRegistered.set(true);
+        if (null != registeredFuture.get()) { //complete the pending future.
```

Comment:
Technically for this to be correct, registeredFuture.get() should be assigned to a local variable and if it is not null complete should be called.